### ISSUE TITLE
PSR12 does not forbid blank lines at the start of the class body

### ISSUE THREAD

**Issue by InvisibleSmiley** [squizlabs/PHP_CodeSniffer#3437]
**Describe the bug**
In certain situations, PSR12 does not forbid blank lines at the start of class bodies, **e.g. with class constants at the top**.
This is in contrast to:
- the definition of PSR-12 (https://www.php-fig.org/psr/psr-12/) that states "Opening braces MUST be on their own line and MUST NOT be preceded or followed by a blank line."
- the comment above `<rule ref="PSR2.Classes.ClassDeclaration"/>` in file PSR12/ruleset.xml which states: "Opening braces MUST be on their own line and MUST NOT be preceded or followed by a blank line."

In some other cases, blank lines at the start of class bodies are rejected by specific rules, e.g. those for trait imports (use), properties and methods (functions).

**Code sample**
```php
<?php

namespace Foo;

class Bar
{


    public const X = 1;
}
```

**Expected behavior**
Blank lines at the start of class bodies should always result in some sort of error with PSR12.

**Versions (please complete the following information):**
 - OS: Windows 10
 - PHP: 7.4
 - PHPCS: 3.6.0
 - Standard: PSR12

**Additional context**
I think the ClassDeclaration sniff neeeds to be adjusted for PSR12 so that it does not matter which kind of element is at the start of the class body (constants, property, method, ...). Leaving this logic to specific sniffs might result in future breakages as PHP evolves.

**Comment by scorgn** [comment]
I am seeing this as well, it seems that it is also not notifying of a blank line after a method opening brace.

> 5. Control Structures
> [...]
> The body MUST be on the next line after the opening brace

I am on
- OS: MacOS Catalina 10.15.7
- PHP: 7.3.30
- PHPCS: 3.6.0
- Standard: PSR12

**Comment by gsherwood** [comment]
This looks to be a missing check from the start. PSR2 allows as many blank lines as you want after the opening brace, so using that sniff alone was not enough to enforce the PSR12 rule. I'll add a new sniff to specifically enforce no blank lines after the opening brace.

**Comment by gsherwood** [comment]
> it seems that it is also not notifying of a blank line after a method opening brace.

I think that's just for control structures, not functions and methods. I can't see a rule under methods that enforces no blank lines at the start of the method body

**Comment by gsherwood** [comment]
I've added `PSR12.Classes.OpeningBraceSpace` to enforce this. The sample code provided now produces this error:
```
FILE: temp.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 6 | ERROR | [x] Opening brace must not be followed by a blank line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```
It is auto-fixed like this:
```diff
--- temp.php
+++ PHP_CodeSniffer
@@ -4,7 +4,5 @@

 class Bar
 {
-
-
     public const X = 1;
 }
```
Thanks for reporting this.
